Added pruning for . and ..

sco1 opened the pull request:
When calling `subdir` with no input arguments we get a tree of the current directory. Since this tree is generated using recursive `dir` calls, if there are a lot of nested subfolders we obtain a lot of `.` and `..` 'directory' entries.

For example, if we run the following code in an empty directory:

```
mkdir('test1/test2');
subdir;
```

We receive the following command window output:

```
C:\test\.
C:\test\..
C:\test\test1
C:\test\test1\.
C:\test\test1\..
C:\test\test1\test2
C:\test\test1\test2\.
C:\test\test1\test2\..
```

If we incorporate the mod, we receive the following command window output:

```
C:\test\test1
C:\test\test1\test2
```

sco1 committed:
Added pruning for . and ..

kakearney commented:
I like the idea, but this code addition drops all Linux-style hidden folders (.ssh, .git, etc.), not just the . and .. ones.  Should be pretty easy to adapt, though, and will do ASAP.

sco1 commented:
Crap, I didn't notice that. I'll have to ponder the most efficient way.

sco1 committed:
Fixed (hopefully) dot filtering

Dot filtering should now keep hidden directories (e.g. .\.git) in the directory list.

sco1 commented:
Perhaps a regex?

```
[~, ~, tail] = cellfun(@fileparts, {Files(:).name}, 'UniformOutput', false);
dottest = cellfun(@(x) isempty(regexp(x, '\.+(\w+$)', 'once')), tail);
Files(dottest & [Files(:).isdir]) = [];
```

I don't have a linux distro to test with but this at least leaves my .git folders alone.

kakearney commented:
Okay, I like that.  Could probably build on that to do more robust directory-filtering, too, which has been on my todo list for a while.  Merging.
